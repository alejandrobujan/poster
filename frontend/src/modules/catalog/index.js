import * as actions from './actions';
import * as actionTypes from './actionTypes';
import reducer from './reducer';
import * as selectors from './selectors';

export {default as SearchBar} from "./components/SearchBar";
export {default as CategorySelector} from './components/CategorySelector';
export {default as Post} from './components/Post';
export {default as Posts} from './components/Posts';
export {default as Feed} from './components/Feed';

// eslint-disable-next-line
export default {actions, actionTypes, reducer, selectors};


