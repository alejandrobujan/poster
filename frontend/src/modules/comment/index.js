import * as actions from './actions';
import reducer from './reducer';
import * as selectors from './selectors';

export {default as Comments} from './components/Comments';
export {default as Comment} from './components/Comment';

// eslint-disable-next-line
export default  {actions, reducer, selectors};