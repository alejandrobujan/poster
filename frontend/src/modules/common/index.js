export {default as Pager} from "./components/Pager";
export {default as Errors} from "./components/Errors";
export {default as BackLink} from "./components/BackLink";
export {default as UserCard} from "./components/UserCard";
export {default as PostLink} from "./components/PostLink";
export {default as ImagesCard} from "./components/ImagesCard";