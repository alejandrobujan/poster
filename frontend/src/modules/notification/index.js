import * as actions from './actions';
import * as actionTypes from './actionTypes';
import reducer from './reducer';
import * as selectors from './selectors';

export {default as Notifications} from "./components/Notifications";
export {default as Notification} from "./components/Notification";

// eslint-disable-next-line
export default {actions, actionTypes, reducer, selectors};


