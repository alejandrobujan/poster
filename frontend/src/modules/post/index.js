import * as actions from './actions';
import reducer from './reducer';
import * as selectors from './selectors';

export {default as CreatePost} from './components/CreatePost';
export {default as CategorySelector} from './components/CategorySelector';
export {default as Post} from './components/Post';
export {default as Posts} from './components/Posts';
export {default as Feed} from './components/Feed';
export {default as PostDetails} from './components/PostDetails';


// eslint-disable-next-line
export default  {actions, reducer, selectors};