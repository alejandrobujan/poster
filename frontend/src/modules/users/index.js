import * as actions from './actions';
import * as actionTypes from './actionTypes';
import reducer from './reducer';
import * as selectors from './selectors';

export {default as Login} from './components/Login';
export {default as SignUp} from './components/SignUp';
export {default as Logout} from './components/Logout';
export {default as UpdateProfile} from './components/UpdateProfile';

// eslint-disable-next-line
export default {actions, actionTypes, reducer, selectors};