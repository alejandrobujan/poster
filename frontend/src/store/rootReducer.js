import {combineReducers} from 'redux';

import app from '../modules/app';
import users from '../modules/users';
import post from '../modules/post';
import catalog from '../modules/catalog';
import notification from '../modules/notification';

const rootReducer = combineReducers({
    app: app.reducer,
    users: users.reducer,
    post: post.reducer,
    catalog: catalog.reducer,
    notification: notification.reducer
});

export default rootReducer;