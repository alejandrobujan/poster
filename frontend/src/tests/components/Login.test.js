import { fireEvent, render, screen, waitFor } from "@testing-library/react";
import React from "react";
import { MemoryRouter } from "react-router-dom";
import {Provider} from 'react-redux';
import renderer from "react-test-renderer";
import store from "../../store";
import { Login, Logout, SignUp } from "../../modules/users";

import {createMemoryHistory} from 'history';

describe("Login", () => {

    it("renders correctly", () => {
		const tree = renderer
        .create(
				<Provider store={store}>
					<MemoryRouter>
						<Login/>
					</MemoryRouter>
				</Provider>
			)
			.toJSON();
		expect(tree).toMatchSnapshot();
	});




	it("do sign up log out and log in", async () => {
		const history = createMemoryHistory();

		render(
			<Provider store={store}>
				<MemoryRouter>
					<SignUp/>
				</MemoryRouter>
			</Provider>
		);

		const signUpLink = screen.getByText(/Sign Up/i);
		
		fireEvent.click(signUpLink);

		await waitFor(() => {
			screen.getByText(/Email/);
		});

		const login = screen.getByLabelText('Login');

		fireEvent.change(login, {
			target: { value: "Pepe" },
		});

		const email = screen.getByLabelText('Email');

		fireEvent.change(email, {
			target: { value: "Pepe@pepe.com" },
		});

		const password = screen.getByLabelText('Password');

		fireEvent.change(password, {
			target: { value: "Pepe" },
		});

		const confirmPassword = screen.getByLabelText('Confirm password');

		fireEvent.change(confirmPassword, {
			target: { value: "Pepe" },
		});

		const firstName = screen.getByLabelText('First name');

		fireEvent.change(firstName, {
			target: { value: "Pepe" },
		});

		const lastName = screen.getByLabelText('Last name');

		fireEvent.change(lastName, {
			target: { value: "Pepe" },
		});

		const submit = screen.getByRole('button', {name: 'Submit'});

		fireEvent.click(submit);

		expect(history.location.pathname).toEqual('/');

		await waitFor(() => {
			screen.queryByText(/No posts found/i);
		});


		//A partir de aqui, aparecen los problemas
		/*const logoutClick = screen.getByTestId('dropdown'); //esto estaba anadido al Header, se ha eliminado


		const userProfile = screen.getByText('Pepe');

		fireEvent.click(userProfile);

		await waitFor(() => {
			screen.getByText(/Logout/);
		});

		const logoutLabel = screen.getByText('Logout');

		fireEvent.click(logoutLabel);

		expect(history.location.pathname).toEqual('/users/logout');*/
		


	});

});