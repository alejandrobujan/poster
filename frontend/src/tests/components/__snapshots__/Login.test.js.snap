// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login renders correctly 1`] = `
<div
  className="container"
>
  <div
    className="card bg-light border-dark m-4"
  >
    <h5
      className="card-header text-center"
    >
      Login
    </h5>
    <div
      className="card-body"
    >
      <form
        className="needs-validation"
        noValidate={true}
        onSubmit={[Function]}
      >
        <div
          className="form-group row"
        >
          <label
            className="col-md-3 col-form-label"
            htmlFor="login"
          >
            Login
          </label>
          <div
            className="col-md-9"
          >
            <input
              autoFocus={true}
              className="form-control"
              id="login"
              onChange={[Function]}
              required={true}
              type="text"
              value=""
            />
            <div
              className="invalid-feedback"
            >
              The login is required
            </div>
          </div>
        </div>
        <div
          className="form-group row"
        >
          <label
            className="col-md-3 col-form-label"
            htmlFor="password"
          >
            Password
          </label>
          <div
            className="col-md-9"
          >
            <input
              className="form-control"
              id="password"
              onChange={[Function]}
              required={true}
              type="password"
              value=""
            />
            <div
              className="invalid-feedback"
            >
              The password is required
            </div>
          </div>
        </div>
        <div
          className="text-center"
        >
          <button
            className="btn btn-primary"
            id="loginButton"
            type="submit"
          >
            Submit
          </button>
        </div>
      </form>
    </div>
  </div>
  <p
    className="text-center"
  >
    Not registered yet?
    <br />
    <a
      href="/poster/users/signup"
      onClick={[Function]}
    >
      Sign up
    </a>
  </p>
</div>
`;
